File: Perl/calculator.pl
#!/usr/bin/perl

use strict;
use warnings;

sub main_menu 
{
    print "\n--- Main Menu ---\n";
    print "1. Addition\n";
    print "2. Subtraction\n";
    print "3. Multiplication\n";
    print "4. Division\n";
    print "5. Exit\n";
    print "Enter your choice: ";
}

sub addition 
{
    print "\nEnter the first number: ";
    my $num1 = <STDIN>;
    chomp($num1);
    print "Enter the second number: ";
    my $num2 = <STDIN>;
    chomp($num2);
    my $result = $num1 + $num2;
    print "$num1 + $num2 = $result\n";
}

sub subtraction 
{
    print "\nEnter the first number: ";
    my $num1 = <STDIN>;
    chomp($num1);
    print "Enter the second number: ";
    my $num2 = <STDIN>;
    chomp($num2);
    my $result = $num1 - $num2;
    print "$num1 - $num2 = $result\n";
}

sub multiplication 
{
    print "\nEnter the first number: ";
    my $num1 = <STDIN>;
    chomp($num1);
    print "Enter the second number: ";
    my $num2 = <STDIN>;
    chomp($num2);
    my $result = $num1 * $num2;
    print "$num1 * $num2 = $result\n";
}

sub division 
{
    print "\nEnter the numerator: ";
    my $num1 = <STDIN>;
    chomp($num1);
    print "Enter the denominator: ";
    my $num2 = <STDIN>;
    chomp($num2);
    if ($num2 == 0) {
        print "Error: division by zero\n";
    } else {
        my $result = $num1 / $num2;
        print "$num1 / $num2 = $result\n";
    }
}

while (1) 
{
    main_menu();
    my $choice = <STDIN>;
    chomp($choice);
    if ($choice == 1) 
    {
        addition();
    } elsif ($choice == 2) 
    {
        subtraction();
    } elsif ($choice == 3) 
    {
        multiplication();
    } elsif ($choice == 4) 
    {
        division();
    } elsif ($choice == 5) 
    {
        last;
    } else 
    {
        print "Invalid choice\n";
    }
}

File: C/Menu Driven Programs/Calculator.c
// Menu Driven Program for a calculator in C
#include <stdio.h>

int main() 
{
    int choice = -1;
    int a, b;

        printf("MENU:\n1. Addition\n2. Subtraction\n3. Multiplication\n4. Division\n5. Exit");
    
    while(1){
        printf("\nChoose your Operation :");
        scanf("%d", &choice);
        
        if(choice == 1){
            printf("Enter First number :");
            scanf("%d", &a);
            printf("Enter Second number :");
            scanf("%d", &b);
            printf("Result : %d + %d = %d\n", a, b, (a+b));
        }
        else if (choice == 2){
            printf("Enter First number :");
            scanf("%d", &a);
            printf("Enter Second number :");
            scanf("%d", &b);
            printf("Result : %d - %d = %d\n", a, b, (a-b));
        }
        else if (choice == 3){
            printf("Enter First number :");
            scanf("%d", &a);
            printf("Enter Second number :");
            scanf("%d", &b);
            printf("Result : %d * %d = %d\n", a, b, (a*b));
        }
        else if (choice == 4){
            printf("Enter First number :");
            scanf("%d", &a);
            printf("Enter Second number :");
            scanf("%d", &b);
            printf("Result : %d / %d = %d\n", a, b, (a/b));
        }
        else if (choice == 5){
            printf("GooDByE...!!!\n");            
            break;
        }
        else{
            printf("Invalid Input\n");
        }
    }
    return 0;
}